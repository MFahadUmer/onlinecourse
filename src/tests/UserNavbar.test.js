import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import UserNavbar from '../component/userNavbar';

describe('User Navbar Test', () => {
  it('should render author panel navbar', () => {
    render(
      <BrowserRouter>
        <UserNavbar />
      </BrowserRouter>,
    );
    expect(screen.getByText('Favourites')).toBeTruthy();
  });

  it('should render author panel navbar', () => {
    render(
      <BrowserRouter>
        <UserNavbar />
      </BrowserRouter>,
    );
    expect(screen.getByText('Favourites').href).toBe('http://localhost/favourites');
  });

  it('matches the Navbar component snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <UserNavbar />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
