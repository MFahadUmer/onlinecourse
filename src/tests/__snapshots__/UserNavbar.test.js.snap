// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Navbar Test matches the Navbar component snapshot 1`] = `
<div
  className="userNavbar"
>
  <div>
    <div
      className="dropdown"
    >
      <span
        className="fas fa-bars userNavbarLink dropdown"
      />
      <div
        className="dropdown-content"
      >
        <div
          className="dropDownLinksContainer"
        >
          <a
            className="dropDownLinks"
            href="/favourites"
            onClick={[Function]}
          >
             Favourites 
          </a>
          <a
            className="dropDownLinks"
            href="https://reactonlinecourse.herokuapp.com/"
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </div>
  <p
    className="userNavbarLink"
  >
    Courses
  </p>
  <p>
    <span
      className="fas fa-search userNavbarLink"
    />
  </p>
</div>
`;
